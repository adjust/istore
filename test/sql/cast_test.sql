SELECT '1=>1, -1=>-1'::istore::bigistore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>-1'::bigistore::istore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3000000000'::bigistore::istore;
ROLLBACK;
