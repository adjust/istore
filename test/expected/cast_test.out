-- casts should cast from istore to bigistore;
SELECT '1=>1, -1=>-1'::istore::bigistore;
      bigistore       
----------------------
 "-1"=>"-1", "1"=>"1"
(1 row)

-- casts should cast from bigistore to istore;
SELECT '1=>1, -1=>-1'::bigistore::istore;
        istore        
----------------------
 "-1"=>"-1", "1"=>"1"
(1 row)

-- casts should fail cast from bigistore to istore if any value is to big;
SELECT '1=>1, -1=>3000000000'::bigistore::istore;
ERROR:  integer out of range
